services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -174,10 +176,16 @@ private Destination getDestinationForTopic(String replyToTopic) {
         }
     }
 
-    private void doDiscoverIslCommand(DiscoverIslCommandData command) {
+    private void doDiscoverIslCommand(CommandMessage message) {
+        DiscoverIslCommandData command = (DiscoverIslCommandData) message.getData();
```

Comment:
Why for move type conversion inside method? The caller must be responsible to supply correct type (in our case caller do "instanceof" and definetely know variable type).

PS CorrelationId can/should be passed as separate argument.